crates/iota-protocol-config/src/lib.rs

```diff
@@ -357,6 +355,10 @@ struct FeatureFlags {
     #[serde(skip_serializing_if = "is_false")]
     move_auth: bool,
 
+    // If true, enables publishing package metadata v1 along with the package.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    publish_package_metadata: Option<u64>,
```

Comment:
```suggestion
    // Package metadata version, defines the package metadata version published with the package.
    #[serde(skip_serializing_if = "Option::is_none")]
    publish_package_metadata_version: Option<u64>,
```
`FeatureFlags` contains only boolean flags, so it probably makes sense to move this property to the `ProtocolConfig`

Reply:
So I would suggest to revert to a `publish_package_metadata: bool` because it is indeed a flag to enable a feature. In the future, we'll have a bool for v2, if necessary, like  `native_charging_v2` and `consensus_linearize_subdag_v2`. From the protocol config perspective, the version is not really important (at least for now). We could handle it internally like `Transaction` or `TransactionEffects`.

Reply:
I would prefer to have a version in the config, but I don't insist; you can decide what option to leave

Reply:
[40b2281](https://github.com/iotaledger/iota/pull/9296/commits/40b22810f79420530d390d12c9a21d251a774535)